kroxylicious-filters/kroxylicious-record-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/RecordEncryptorTest.java

```diff
@@ -26,76 +25,103 @@
 
-    private static Dek.Decryptor DECRYPTOR;
+    private static TestComponents components;
 
-    private static Dek.Encryptor ENCRYPTOR;
-    private static ByteBuffer EDEK;
-    private static Serde<ByteBuffer> EDEK_SERDE;
+    record TestComponents(
+                          Dek.Encryptor encryptor,
+                          Dek.Decryptor decryptor,
+                          Serde<ByteBuffer> edekSerde) {}
 
     @BeforeAll
-    public static void initKeyContext() throws NoSuchAlgorithmException, ReflectiveOperationException {
-        var generator = KeyGenerator.getInstance("AES");
-        var key = generator.generateKey();
-        EDEK = ByteBuffer.wrap(key.getEncoded()); // it doesn't matter for this test that it's not encrypted
-
-        var dek = mock(Dek.class);
-        doReturn(EDEK).when(dek).edek();
-        var encryptorConstructor = Dek.Encryptor.class.getDeclaredConstructor(Dek.class, CipherSpec.class, SecretKey.class, Integer.TYPE);
-        encryptorConstructor.setAccessible(true);
-        ENCRYPTOR = encryptorConstructor.newInstance(dek, CipherSpec.AES_128_GCM_128, key, 1_000_000);
-        doReturn(ENCRYPTOR).when(dek).encryptor(anyInt());
-
-        var decryptorConstructor = Dek.Decryptor.class.getDeclaredConstructor(Dek.class, CipherSpec.class, SecretKey.class);
-        decryptorConstructor.setAccessible(true);
-        DECRYPTOR = decryptorConstructor.newInstance(dek, CipherSpec.AES_128_GCM_128, key);
-        doReturn(DECRYPTOR).when(dek).decryptor();
-
-        EDEK_SERDE = new Serde<ByteBuffer>() {
-            @Override
-            public int sizeOf(ByteBuffer object) {
-                return object.remaining();
-            }
-
-            @Override
-            public void serialize(
-                                  ByteBuffer object,
-                                  @NonNull ByteBuffer buffer) {
-                var p0 = object.position();
-                buffer.put(object);
-                object.position(p0);
-            }
-
-            @Override
-            public ByteBuffer deserialize(@NonNull ByteBuffer buffer) {
-                throw new UnsupportedOperationException();
-            }
-        };
+    public static void initKeyContext() {
+        components = setup(256);
     }
 
-    private Record encryptSingleRecord(Set<RecordField> fields, long offset, long timestamp, String key, String value, Header... headers) {
+    static TestComponents setup(int keysize) {
+        try {
+            var generator = KeyGenerator.getInstance("AES");
+            generator.init(keysize);
+            var key = generator.generateKey();
+            var EDEK = ByteBuffer.wrap(key.getEncoded()); // it doesn't matter for this test that it's not encrypted
+
+            var dek = mock(Dek.class);
+            doReturn(EDEK).when(dek).edek();
+            var encryptorConstructor = Dek.Encryptor.class.getDeclaredConstructor(Dek.class, CipherSpec.class, SecretKey.class, Integer.TYPE);
```

Comment:
Having to rely on reflection is a code smell.  It'll also make refactorings harder.  Would it be better to make the ctor `@visiblefortesting` (or a factory)

Reply:
I've refactored it to not use reflection and instead use a DekManager with a KMS service that offers a single fixed DEK to obtain the dek/encryptor/decryptor from.